MODULE.bazel

```diff
@@ -1,5 +1,10 @@
 """Bazel build and test dependencies."""
 
+# NOTE: When editing this file, also update the lockfile.
+#   bazel mod deps --lockfile_mode=update
+#   bazel build //src/test/tools/bzlmod:update_default_lock_file
+#   bazel-bin/src/test/tools/bzlmod/update_default_lock_file
```

Comment:
Could this just be `bazel run` instead of `build` plus manual run? This may also only be needed when `MODULE.tools` is changed - cc @meteorcloudy .

Reply:
Yes, this is only needed when `MODULE.tools` is changed.

And you actually have to use `bazel run //src/test/tools/bzlmod:update_default_lock_file` since it needs `BUILD_WORKING_DIRECTORY` to be set.

Reply:
When I tried `bazel run`:
```
$ bazel run //src/test/tools/bzlmod:update_default_lock_file
INFO: Analyzed target //src/test/tools/bzlmod:update_default_lock_file (0 packages loaded, 1 target configured).
INFO: Found 1 target...
Target //src/test/tools/bzlmod:update_default_lock_file up-to-date:
  bazel-bin/src/test/tools/bzlmod/update_default_lock_file
INFO: Elapsed time: 0.381s, Critical Path: 0.01s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/src/test/tools/bzlmod/update_default_lock_file
Running: /usr/local/google/home/jcater/.cache/bazel/_bazel_jcater/a85a0dacd9e20e18c47928db0f1530e0/execroot/_main/bazel-out/k8-fastbuild/bin/src/bazel mod deps
Extracting Bazel installation...
Starting local Bazel server and connecting to it...
WARNING: --enable_bzlmod is set, but no MODULE.bazel file was found at the workspace root. Bazel will create an empty MODULE.bazel file. Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel. For more details, please refer to https://github.com/bazelbuild/bazel/issues/18958.
<root> (@_)
```

But it did update `src/test/tools/bzlmod/MODULE.bazel.lock`.


When I split into `bazel build .. && ...`:
```
 bazel build //src/test/tools/bzlmod:update_default_lock_file && bazel-bin/src/test/tools/bzlmod/update_default_lock_file
INFO: Analyzed target //src/test/tools/bzlmod:update_default_lock_file (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
Target //src/test/tools/bzlmod:update_default_lock_file up-to-date:
  bazel-bin/src/test/tools/bzlmod/update_default_lock_file
INFO: Elapsed time: 0.343s, Critical Path: 0.01s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
```

No error, but `src/test/tools/bzlmod/MODULE.bazel.lock` wasn't changed.

This is incredibly confusing.

Reply:
The bash runfile libraries seems to fail with no error message when directly running the bash binary. @fmeum Am I using the runfiles library correctly? https://github.com/bazelbuild/bazel/blob/013683f6364dd15209aee83a71df58e3bfb4aff1/src/test/tools/bzlmod/update_default_lock_file.sh#L4

Reply:
You are, this is a bug. I will submit a fix.

Reply:
A fix for this issue has been included in [Bazel 7.0.0 RC2](https://releases.bazel.build/7.0.0/rc2/index.html). Please test out the release candidate and report any issues as soon as possible. Thanks!